### PR TITLE
ci: More maintainable code

### PR DESCRIPTION
**Pull request by NickeZ**
Mantaining a json-blob is not as easy as maintaining a yaml multiline string. I don't think icon_emoji and props/card pull their weight for us to use the "PAYOLOAD" field instead of the "TEXT" field.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -199,24 +199,23 @@ jobs:
...
         with:
           MATTERMOST_WEBHOOK_URL: ${{ secrets.MM_WEBHOOK_URL }}
-          PAYLOAD: |-
-            {
-              "channel": "eng-artifacts",
-              "text": "**New artifacts built**\n([${{ github.ref_name }}](https://github.com/${{ github.repository }}/tree/${{ github.ref_name }}), [${{ steps.vars.outputs.git_sha_short }}](https://github.com/${{ github.repository }}/commit/${{ github.sha }}), [artifacts](https://github.com/${{ github.repository }}/actions/runs/${{github.run_id}}#artifacts))\n* Android - [APK](${{needs.android.outputs.artifact-url}})\n* Linux - [AppImage](${{needs.qt-linux.outputs.artifact-url-ai}}), [DEB](${{needs.qt-linux.outputs.artifact-url-deb}}), [RPM](${{needs.qt-linux.outputs.artifact-url-rpm}})\n* MacOS - [Zip](${{needs.macos.outputs.artifact-url}})",
-              "icon_emoji": "white_check_mark",
-              "props": {"card": "Repository: [${{ github.repository }}](https://github.com/${{ github.repository }})"}
-            }
+          MATTERMOST_ICON_URL: https://cdn4.iconfinder.com/data/icons/basic-ui-2-line/32/check-mark-checklist-complete-done-512.png
```
Comment (benma) [inline .github/workflows/ci.yml:202]: Will this be downloaded/called every time anyone views the MM eng-artifacts channel? Or does the MM server download it once and store it on the server?

Comment (NickeZ) [reply .github/workflows/ci.yml:202]: Your browser should cache it for a year according to the headers (cache-control: public, max-age=31536000)

In their example they do the same:

https://github.com/mattermost/action-mattermost-notify?tab=readme-ov-file#usage

Comment (NickeZ) [reply .github/workflows/ci.yml:202]: From what I can see in mattermost in the developer console, it seems the server could cache it. This is what is being fetched in the client:

`https://most.teamshift.ch/api/v4/image?url=https%3A%2F%2Fcdn4.iconfinder.com%2Fdata%2Ficons%2Fbasic-ui-2-line%2F32%2Fcheck-mark-checklist-complete-done-512.png`

**Review by benma — APPROVED** [review]
utACK

**Pull request merged**